src/test/FsCheck.Fake/TestParallelBuildOrder.fs

```diff
@@ -0,0 +1,143 @@
+﻿module FsCheck.Fake.TestParallelBuildOrder
+
+open System
+open System.Collections.Generic
+open global.Xunit
+open Fake
+open FsCheck
+
+let inline (==>) a b = global.Fake.AdditionalSyntax.(==>) a b
+
+// checks whether the given order is consistent with all dependencies
+let validateBuildOrder (order : list<Target[]>) (rootTarget : string) =
+    let rootTarget = getTarget rootTarget
+
+    // store a "level" for all targets in order
+    let targetLevelMap = 
+        order |> List.mapi (fun i arr ->
+                arr |> Array.map (fun t -> t.Name, i)
+              )
+              |> Seq.concat
+              |> Map.ofSeq
+
+    // check whether the assigned level is smaller or equal to
+    // the given max level.
+    let checkLevel (target : Target) (maxLevel : int) =
+        match Map.tryFind target.Name targetLevelMap with
+            | Some level ->
+                if level > maxLevel then
+                    failwithf "found target on unexpected level %d (should be at most %d)" level maxLevel
+
+            | None ->
+                failwithf "target %A was not assigned a level but occurs in the dependency tree" target.Name
+    
+    // recursively validate the target levels
+    let rec validate (t : Target) (maxLevel : int) =
+        checkLevel t maxLevel
+        let realLevel = Map.find t.Name targetLevelMap
+
+        let deps = t.Dependencies |> List.map getTarget
+        for d in deps do
+            validate d (realLevel - 1)
+
+    // initially the max-level is unbounded
+    validate rootTarget Int32.MaxValue
+
+[<Fact>]
+let ``Independent targets are parallel``() =
+    TargetDict.Clear()
+
+    Target "a" DoNothing
+    Target "b" DoNothing
+    Target "c" DoNothing
+    
+    Target "dep" DoNothing
+
+    "a" ==> "dep" |> ignore
+    "b" ==> "dep" |> ignore
+    "c" ==> "dep" |> ignore
+
+    let order = determineBuildOrder "dep"
+
+    validateBuildOrder order "dep"
+
+    match order with
+        | [f; [|d|]] ->
```

Comment:
Maybe it's worth to introduce an active pattern `Target`, which'd enable syntax like

```fsharp
match order with
| [Target "dep"; [| Target "a"; Target "b"; Target "c" |] -> ...
| _ -> failwith ...
```